Author: Luca Boccassi <bluca@debian.org>
Description: fix ip vrf exec with libbpf
 The size of bpf_insn is passed to bpf_load_program instead of the number
 of elements as it expects, so ip vrf exec fails with:
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=980299
Forwarded: yes, https://www.spinics.net/lists/netdev/msg714642.html
--- a/lib/bpf_glue.c
+++ b/lib/bpf_glue.c
@@ -14,7 +14,8 @@
 		     size_t size_log)
 {
 #ifdef HAVE_LIBBPF
-	return bpf_load_program(type, insns, size_insns, license, 0, log, size_log);
+	return bpf_load_program(type, insns, size_insns / sizeof(struct bpf_insn),
+				license, 0, log, size_log);
 #else
 	return bpf_prog_load_dev(type, insns, size_insns, license, 0, log, size_log);
 #endif
